docs/SETUP_TLS_CERTIFICATES.md

```diff
@@ -0,0 +1,232 @@
+1. Generate root CA certificate:
+    ```
+    certstrap init --cn "snaptest-ca" --o "snaptest" --ou "ca" --key-bits 2048 --years 1 --passphrase '<ca-secure-passphrase>'
+    ```
+    Relevant files:
+    * `./out/snaptest-ca.key` - CA's private key, encrypted with `<ca-secure-passphrase>`; needs to be protected from unauthorized access,
+    * `./out/snaptest-ca.crt` - CA's public certificate, (a certificate root).
+1. Generate server certificate and key to use by plugins:
+    ```
+    certstrap request-cert --cn "snaptest-srv" --ip "127.0.0.1" --domain "localhost" --passphrase '' --key-bits 2048 --o "snaptest" --ou "server"
+    certstrap sign "snaptest-srv" --CA "snaptest-ca" --passphrase '<ca-secure-passphrase>' --years 1
+    ```
+    This will produce a private key certificate file necessary to establish secure TLS channel on server. Server's private key file should not be passphrase-protected (or each plugin would have to ask for passphrase on start).
+    Relevant files:
+    * `./out/snaptest-srv.key` - server's private key,
+    * `./out/snaptest-srv.crt` - server's public certificate.
+1. Generate client certificate and key to use by framework:
+    ```
+    certstrap request-cert --cn "snaptest-cli" --ip "127.0.0.1" --domain "localhost" --passphrase '' --key-bits 2048 --o "snaptest" --ou "client"
+    certstrap sign "snaptest-cli" --CA "snaptest-ca" --passphrase '<ca-secure-passphrase>' --years 1
+    ```
+    This will produce a private key certificate file necessary to securely connect to plugins. Client's private key file should not be passphrase-protected.
+    Relevant files:
+    * `./out/snaptest-cli.key` - client's private key,
+    * `./out/snaptest-cli.crt` - client's public certificate.
+    
+Files that need to be given at plugin startup:
+* `snaptest-ca.crt` - CA's certificate,
+* `snaptest-srv.key` - server's private key file,
+* `snaptest-srv.crt` - server's certificate file.
+
+Files that need to be given at `snapteld` startup:
+* `snaptest-ca.crt` - CA's certificate,
+* `snaptest-cli.key` - framework's private key file,
+* `snaptest-cli.crt` - framework's certificate file.
+
+## Obtaining legitimate certificate from a public CA Let's Encrypt
+
+This section describes how to obtain a certificate from CA authority. Example will be based on [Let's encrypt](https://letsencrypt.org/) public CA service.
+
+In general, obtaining a TLS certificate involves:
+* on customer side: issuing request to CA authority, specifying domain name and contact information,
+* on CA side: validating the customer's title to the requested domain.
+
+This guide will show how to obtain certificate for sample domains.
+
+### Prerequisites
+
+There's no way around registering a physical domain. In this walkthrough we'll assume that domains `frodo.buthey.net` and `bilbo.buthey.net` were registered and they're mapped to actual machines (VM or physical).
```

Comment:
Wording. Maybe: 

> There's no way around the need to register a domain name to use this example. In this walkthrough we'll assume that domains `frodo.buthey.net` and `bilbo.buthey.net` were registered and they're DNS records map to actual machines (VM or physical).

Note that I changed "mapped" to be more specific. You can change what I wrote, but I think being more specific is helpful.